sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecution.scala

```diff
@@ -35,27 +35,49 @@ class SqlScriptingExecution(
     session: SparkSession,
     args: Map[String, Expression]) extends Iterator[DataFrame] {
 
-  // Build the execution plan for the script.
-  private val executionPlan: Iterator[CompoundStatementExec] =
-    SqlScriptingInterpreter(session).buildExecutionPlan(sqlScript, args)
+  private val interpreter = SqlScriptingInterpreter(session)
 
-  private var current = getNextResult
+  // Frames to keep what is being executed.
+  private val context: SqlScriptingExecutionContext = {
+    val ctx = new SqlScriptingExecutionContext()
+    interpreter.buildExecutionPlan(sqlScript, args, ctx)
+    ctx
```

Comment:
passing `ctx` to interpreter and doing a part of initialization there doesn't make a lot of sense to me.
can we do something like `ctx.frames.addOne(new SqlScriptingExecutionFrame(iter))` here where `iter` is `buildExecutionPlan` output ?

Reply:
Context is now passed just to be propagated further down the stack.